### PR TITLE
[MNT] Updated and added tests to cover newly added functionalitie

### LINKED ISSUE #106: Update e2e test to cover newly added functionalities

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #106 
- Closes https://github.com/neurobagel/planning/issues/271

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [x] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

## Summary by Sourcery

Enhance data store handling, improve UI component structure, and expand test coverage

Enhancements:
- Use Immer in updateColumnMissingValues to automatically adjust categorical levels when marking values as missing
- Introduce createSeededUuidGenerator using seedrandom for deterministic UUID generation in MultiColumnMeasures
- Refactor ColumnTypeCollapse to default export and extract ExpandableSection into a standalone component
- Add data-cy attributes to Categorical, Download, and ValueAnnotationTabs components for more reliable test selectors

Build:
- Add seedrandom and @types/seedrandom dependencies to support seeded UUID generation

Tests:
- Extend MainUserFlow E2E tests to cover new columns (subject ID, diagnosis, IQ), multi-column measures, units, missing values, and annotation flows
- Add Cypress component tests for SideColumnNavBar, FileUploader, MultiColumnMeasures, ValueAnnotationTabs, ColumnTypeCollapse, MissingValueButton, and ExpandableSection
- Add store tests for processing data tables with empty lines, missing values logic, and retrieving mapped standardized variables

### CODE REVIEW

File: src/components/MultiColumnMeasures.tsx [hunk 1]
```diff
@@ -22,12 +22,12 @@ interface MultiColumnMeasuresProps {
...
 
 const defaultProps = {
-  generateID: uuidv4,
+  generateID: createSeededUuidGenerator('seed'),
   terms: assessmentTerms,
 };
 
 function MultiColumnMeasures({
-  generateID = uuidv4,
+  generateID = createSeededUuidGenerator('seed'),
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply src/components/MultiColumnMeasures.tsx:30]: This is mostly for testing but there is no issue with the ids being fixed across the app as well as long as they're unique.

File: src/utils/util.ts [hunk 2]
```diff
@@ -1,7 +1,22 @@
+import seedrandom from 'seedrandom';
+import { v4 as uuidv4 } from 'uuid';
 import { Term, TermCard, Columns } from './types';
 
 // Utility functions for MultiColumnMeasures component
 
+// Generate a UUID using a seeded random number generator
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply src/utils/util.ts:7]: So we can reliably use the consistent, unique IDs for testing.

File: cypress/component/ColumnTypeCollapse.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  it('renders the component correctly', () => {
+    cy.mount(
+      <ColumnTypeCollapse
+        dataType={props.dataType}
+        standardizedVariable={props.standardizedVariable}
+        columns={props.columns}
+        onSelect={props.onSelect}
+        selectedColumnId={props.selectedColumnId}
+      />
+    );
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply cypress/component/ColumnTypeCollapse.cy.tsx:25]: > Why are you handling these different scenarios in a single test? 

because they are all "rendering different versions of the component correctly" I don't think more granularity is needed this is just to test the conditions inside the component work as expected given different props.

File: cypress/e2e/MainUserFlow.cy.ts [hunk 4]
```diff
@@ -218,16 +282,46 @@ describe('Main user flow', () => {
...
+    cy.get('[data-cy="4-HC-term-dropdown"] input').should('have.value', 'Healthy Control');
+    cy.get('[data-cy="4-PD-description"]').should('contain', 'Parkinsons');
+    cy.get('[data-cy="4-PD-term-dropdown"] input').should(
+      'have.value',
+      'Parkinsonism caused by methanol'
+    );
+    cy.get('[data-cy="next-button"]').click();
+
+    // Download view
+    cy.get('[data-cy="complete-annotations-alert"]').should('be.visible');
```
[the remark this answers is not in the document]
Comment (rmanaem) [reply cypress/e2e/MainUserFlow.cy.ts:325]: We have those couple of lines above I'll add a couple more checks here as well.
